apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -135,6 +135,7 @@ type PlacementSpec struct {
 	// Policy defines how to select member clusters to place the selected resources.
 	// If unspecified, all the joined member clusters are selected.
 	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="!(has(self.placementType) && has(oldSelf.placementType) && self.placementType != oldSelf.placementType)",message="placement type is immutable"
```

Comment:
do we need has(self.placementType) && has(oldSelf.placementType)  part?

PlacementType is not an optional field.